pkgs/misc/vim-plugins/update.py

```diff
@@ -111,6 +118,22 @@ def latest_commit(self) -> Tuple[str, datetime]:
             updated = datetime.strptime(updated_tag.text, "%Y-%m-%dT%H:%M:%SZ")
             return Path(str(url.path)).name, updated
 
+    def check_for_redirect(self, url: str, req: http.client.HTTPResponse):
+        response_url = req.geturl()
+        if url != response_url:
+            new_owner, new_name = (
+                urllib.parse.urlsplit(response_url).path.strip("/").split("/")[:2]
+            )
+            end_line = f" as {self.alias}\n" if self.alias else "\n"
+            plugin_line = "{owner}/{name}" + end_line
```

Comment:
The `"\n"` should probably be separate since it is always appended. For example:

```python
alias_spec = f" as {self.alias}\n" if self.alias else ""
plugin_line = "{owner}/{name}" + alias_spec + "\n"
```

Reply:
In your example if `self.alias` is truthy the resulting `plugin_line` would result in an extra newline: "{owner}/{name}\n\n".

Reply:
Oh yes, I didn't mean to include the `\n` in the `alias_spec`. Also now that you mention it, I would rather not rely on things being "truthy" but instead check explicitly (`is None`).

Reply:
`alias` is actually an empty string in this case, the truthiness of which is relied on in a couple other `alias or` clauses in this script. Do you prefer `if self.alias != ""`?

Reply:
Yes, I definitely prefer explicit comparisons. "Truthiness" seems to directly contradict the python ethos of explicitness. It reduces readability, especially for people not familiar with python ("truthiness" differs between languages).

Reply:
Done